src/createStorage.js

```diff
@@ -4,37 +4,60 @@ export default function createStorage(type: 'local' | 'sync') {
   return {
     getItem(key: string) {
       return new Promise((resolve, reject) => {
-        chrome.storage[type].get(key, value => {
-          if (chrome.runtime.lastError == null) {
-            // Chrome Storage returns the value in an Object of with its original key. Unwrap the
-            // value from the returned Object to match the `getItem` API.
-            resolve(value[key]);
-          } else {
-            reject();
-          }
-        });
+        if (typeof chrome !== 'undefined') {
+          chrome.storage[type].get(key, (value) => {
+            if (chrome.runtime.lastError == null) {
+              // Chrome Storage returns the value in an Object of with its original key. Unwrap the
+              // value from the returned Object to match the `getItem` API.
+              resolve(value[key]);
+            } else {
+              reject();
+            }
+          });
+        } else {
+          browser.storage[type]
+            .get(key)
+            .then((value) => {
+              resolve(value[key]);
+            })
+            .catch((e) => reject(e));
+        }
       });
     },
     removeItem(key: string) {
       return new Promise((resolve, reject) => {
-        chrome.storage[type].remove(key, () => {
-          if (chrome.runtime.lastError == null) {
-            resolve();
-          } else {
-            reject();
-          }
-        });
+        if (typeof chrome !== 'undefined') {
+          chrome.storage[type].remove(key, () => {
+            if (chrome.runtime.lastError == null) {
+              resolve();
+            } else {
+              reject();
+            }
+          });
+        } else {
+          browser.storage[type]
+            .remove(key)
+            .then(resolve)
+            .catch((e) => reject(e));
```

Comment:
can drop the extra function:

```suggestion
            .catch(reject);
```